samples/excel/90-just-for-fun/gradient.yaml

```diff
@@ -0,0 +1,181 @@
+id: excel-90-just-for-fun-gradient-
+name: Gradient
+description: Uses range formatting and external libraries to draw a colorful gradient within a range. Contributed by Alexander Zlatkovski.
+author: AlexanderZlatkovski
+host: EXCEL
+api_set:
+    ExcelApi: 1.2
+script:
+    content: |-
+        initializeColorPickers();
+
+        // Set up the click handler:
+        const $drawButton = $("#draw-gradient").click(drawGradient);
+        $("#random").click(randomizeColors);
+
+        /** Click-handler for drawing the gradient */
+        async function drawGradient() {
+            $drawButton.prop("disabled", true);
+
+            try {
+                await Excel.run(drawGradientHelper);
+            }
+            catch (error) {
+                OfficeHelpers.Utilities.log(error);
+            }
+
+            $drawButton.prop("disabled", false);
+        }
+
+        /** Helper function to do the actual gradient-drawing */
+        async function drawGradientHelper(context: Excel.RequestContext) {
+            const sheet = await OfficeHelpers.ExcelUtilities
+                .forceCreateSheet(context.workbook, "Gradient", true /*clearOnly*/);
+
+            sheet.getRange().format.set({
+                rowHeight: 15,
+                columnWidth: 15,
+                fill: {
+                    color: "#1e1e1e"
+                }
+            });
+
+            let originalSize = parseInt($('#size').val());
+
+            // Create a 2D in-memory array to hold the colors
+            const colors2D = Array(originalSize);
+            for (let row = 0; row < originalSize; row++) {
+                colors2D[row] = Array(originalSize);
+            }
+
+            fillInColors(colors2D, originalSize);
+
+            let range = sheet.getCell(1, 1).getResizedRange(
+                originalSize - 1, originalSize - 1);
+            for (let row = 0; row < originalSize; row++) {
+                for (let col = 0; col < originalSize; col++) {
+                    range.getCell(row, col).format.fill.color =
+                        colors2D[row][col];
+                }
+            }
+
+            sheet.activate();
```

Comment:
Creating a new sheet and activating it with the same context.sync() will cause an issue in Excel Online. Insert a context.sync() right before sheet.activate().